### PR TITLE
Export graphs to .dot files.

### PR DESCRIPTION
**Pull request by amackworth**
Quick and dirty exporting of graphs to `.dot` files. :sparkles: 

_See #33._

### CODE REVIEW

File: TesseractCore/Exporting.swift [hunk 1]
```diff
@@ -0,0 +1,30 @@
+//
+//  Exporting.swift
+//  TesseractCore
+//
+//  Created by Alexander Mackworth on 2/4/15.
+//  Copyright (c) 2015 Rob Rix. All rights reserved.
+//
+
+public func concat(strings: [String]) -> String {
```
Comment (robrix) [inline TesseractCore/Exporting.swift]: Could this be `private`, please?

File: TesseractCore/Exporting.swift [hunk 2]
```diff
@@ -0,0 +1,30 @@
+//
+//  Exporting.swift
+//  TesseractCore
+//
+//  Created by Alexander Mackworth on 2/4/15.
+//  Copyright (c) 2015 Rob Rix. All rights reserved.
```
Comment (robrix) [inline TesseractCore/Exporting.swift]: Could you use the same header format as the other files? (Basically the copyright line, none of the others.)

File: TesseractCore/Exporting.swift [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    let result = export(graph)
+    let filehandle = NSFileHandle(forWritingAtPath: filename)
+    if let data = result.dataUsingEncoding(NSUTF8StringEncoding) {
+        filehandle?.writeData(data)
+    }
+}
+
+// MARK: - Imports
+import Prelude
+import Foundation
```
Comment (robrix) [inline TesseractCore/Exporting.swift]: Newline at EOF please :smile:

**Comment by amackworth** [comment]
There ya go! :smile_cat:

File: TesseractCore/Exporting.swift [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    return result
+}
+
+public func exportToFile<T>(graph: Graph<T>, filename: String) {
+    let result = export(graph)
+    let filehandle = NSFileHandle(forWritingAtPath: filename)
+    if let data = result.dataUsingEncoding(NSUTF8StringEncoding) {
+        filehandle?.writeData(data)
+    }
+}
```
Comment (robrix) [inline TesseractCore/Exporting.swift]: Upon reflection, I think I’d prefer to keep file handling out of this library altogether—it needn’t concern itself with path access, `URL` vs `String`, etc. I think `export()` above will suffice for that purpose.

**Comment by robrix** [comment]
This. Is. Fantastic! Thank you so much!

There’s a few other minor formatting issues, but they’re tiny little things and rather than go back and forth on them I figure I’ll fix them up after merge :+1:

File: TesseractCore/Exporting.swift (hunk 4 above)
Comment (amackworth) [inline TesseractCore/Exporting.swift]: Mm, yeah, I was debating this myself. I'll just rip this out.

**Comment by robrix** [comment]
I’ll do the `exportToFile` bit too, actually—no need to hold this up.

**Comment by robrix** [comment]
:heart: Thank you so much!

**Comment by amackworth** [comment]
Ha, no problem! I've been working on visualizing my ID3 decision trees in Haskell, actually. Glad I was able to help! :pray: 

P.S. I'd be totally up to work on the decoding as well!

**Comment by robrix** [comment]
That’d be lovely! Thank you so much! :bow:

**Pull request merged**